### ISSUE TITLE
why are there weird red margins of my loaded images?

### ISSUE THREAD

**Issue by elenacliu** [nerfstudio-project/nerfstudio#2318]
**Describe the bug**
When I load the images, they have red margins.

**To Reproduce**
I just install the latest version of nerfstudio (commitid 9b6010ea), and pip install -e . to install it.

**Expected behavior**

**Screenshots**
<img width="508" alt="image" src="https://github.com/nerfstudio-project/nerfstudio/assets/44845151/5ae36028-17e7-4683-b1a4-cec33f24dcf6">

**Additional context**

**Comment by brentyi** [comment]
Hm, I'm seeing something similar:
![image](https://github.com/nerfstudio-project/nerfstudio/assets/6992947/981f260a-8dfa-4ea2-8e95-8106308df7de)

**Comment by tancik** [comment]
I believe those are the poses used for eval.

**Comment by elenacliu** [comment]
I use the instant-ngp data parser, so all of the images are used for eval by default? 
When I use nerfstudio data parser, only a few images have red margins.

**Comment by tancik** [comment]
@FrederikWarburg any idea what's going on?

**Comment by FrederikWarburg** [comment]
Hi! 

I had a look at this. It seems like `InstantNGP(DataParser)` is using all frames for training and for eval. `def _generate_dataparser_outputs(self, split="train"):` takes split as input, but the variable is unused. Therefore, the viewer first plots all the training images without a border (aka all images), and then, on top of the training images, plots the eval images with a red border (aka also all images). 

So, I guess we could 
1) add train/eval split to the `InstantNGP(DataParser) `
2) maybe remove `InstantNGP(DataParser)` as it seems to be almost the same as `Nerfstudio(DataParser)`. I guess the main difference is `get_focal_lengths`, but could easily be added to `Nerfstudio(DataParser)`

In short, I don't think the problem is the recent commit, but rather that using `InstantNGP(DataParser)` reuses all training as eval images. Wouldn't mind doing any of these changes, so let me know what you think is the best way forward?

**Comment by elenacliu** [comment]
@FrederikWarburg thank you for your reply. I may prefer the first way, because many of my scripts just use `instant-ngp-data`, and I don't need so many complex preprocess in other dataparsers.
